### ISSUE TITLE
The ObjectStateManager does not contain an ObjectStateEntry with a reference to an object of type 'TPMS.Shared.Entities.EmailAddress'.

### ISSUE THREAD

**Issue by Marcel-v-B** [zzzprojects/EntityFramework-Extensions#267]
### Description

I get this error on saving an emailaddress. Saving other entities is working fine and saving the same address with AddRange / BulkSaveChanges is working fine also (but with a worse performance).

var emails = new List<EmailAddress>();
emails.Add(new EmailAddress{ Value = "test@test.nl" } );
this.BulkInsert(emails);

### Exception
The ObjectStateManager does not contain an ObjectStateEntry with a reference to an object of type 'TPMS.Shared.Entities.EmailAddress'.

Exception message:
Stack trace:

```txt
bij System.Data.Entity.Core.Objects.ObjectStateManager.GetObjectStateEntry(Object entity)   
bij .[](DbContext this, List`1 ,  , SchemaEntityType , Int32 , List`1 , Boolean )   
bij DbContextExtensions.`1.( )   bij .Read()   bij .()   bij .( , DbCommand )   
bij .( , DbCommand )   bij .Execute(List`1 actions)   
bij .(List`1 )   
bij Z.BulkOperations.BulkOperation.Execute()   
bij Z.EntityFramework.Extensions.EntityBulkOperation`1.BulkInsert()   
bij .[](BulkOperation`1 this, DbContext , List`1 , Boolean , List`1 , Type , String )   
bij .`1.(SchemaEntityType )   
bij System.Collections.Generic.List`1.ForEach(Action`1 action)   
bij .[](DbContext this, BulkOperation`1 , IEnumerable`1 , List`1 )   
bij .[](BulkOperation`1 this, DbContext , IEnumerable`1 , List`1 )   
bij DbContextExtensions.BulkInsert[T](DbContext this, IEnumerable`1 entities, Action`1 bulkOperationFactory)   
bij DbContextExtensions.BulkInsert[T](DbContext this, IEnumerable`1 entities)   
bij UserQuery.RunUserAuthoredQuery() in C:\Users\marce\AppData\Local\Temp\LINQPad5\_nehjcupb\query_gfaazm.cs:regel 54   
bij LINQPad.ExecutionModel.ClrQueryRunner.Run()   
bij LINQPad.ExecutionModel.Server.RunQuery(QueryRunner runner)   
bij LINQPad.ExecutionModel.Server.StartQuery(QueryRunner runner)   
bij LINQPad.ExecutionModel.Server.<>c__DisplayClass153_0.<ExecuteClrQuery>b__0()   
bij LINQPad.ExecutionModel.Server.SingleThreadExecuter.Work()   
bij System.Threading.ThreadHelper.ThreadStart_Context(Object state)   
bij System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)   
bij System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)   
bij System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)   
bij System.Threading.ThreadHelper.ThreadStart()
```

**Comment by Marcel-v-B** [comment]
We are doing some extra work on detecting changes in our custom context, I think this is fighting with EFE because entities are expected to by in the change tracker and I guess EFE is overruling the change tracker.

**Comment by JonathanMagnan** [comment]
Hello @Marcel-v-B ,

It's possible to provide a runnable solution on your side? You can send it in private here: info@zzzprojects.com

If you have some custom code in the detect change, then maybe our library has something that's not supported yet with your context.

Best Regards,

Jonathan

**Comment by JonathanMagnan** [comment]
Hello @Marcel-v-B ,

I believe we never got the project.

Can we close this issue?

Best Regards,

Jonathan

**Comment by Marcel-v-B** [comment]
Hi Jonathan,

For now we found a solution to work around it. The problem still exists,
but I did not have the time yet to isolate it in a project. So yes, it's
better to close it and I'll get back on it later.

Een fijne dag!

Marcel van Beuzekom

Apada
Postbus 27  8090 AA Wezep
t: 038 - 33 75 152   m: 06 - 27 050 817

Marcel@Apada.nl    www.Apada.nl

Op wo 4 sep. 2019 23:48 schreef Jonathan Magnan <notifications@github.com>:
